packages/runtime/container-runtime/src/deltaManagerProxies.ts

```diff
@@ -99,29 +154,63 @@ export abstract class BaseDeltaManagerProxy
 		return this.deltaManager.readOnlyInfo;
 	}
 
-	constructor(protected readonly deltaManager: IDeltaManagerFull) {
+	private readonly eventHandlers: BaseDeltaManagerProxyEventHandlers;
+
+	constructor(
+		protected readonly deltaManager: IDeltaManagerFull,
+		overrides?: Partial<BaseDeltaManagerProxyEventHandlers>,
+	) {
 		super();
 
+		this.eventHandlers = {
+			onPrepareSend: (messageBuffer: unknown[]): void => {
```

Comment:
You already typed it all but I would have done `...args` for all of these to avoid having to type out the types.